### ISSUE TITLE
Make allow-list federation into a top level admin UI toggle

### ISSUE THREAD

**Issue by dariusk** [hometown-fork/hometown#1220]
We should investigate turning `LIMITED_FEDERATION_MODE` into something you can toggle on the admin interface. This would also need to kick off a `tootctl` process as discussed here: https://docs.joinmastodon.org/admin/config/#limited_federation_mode

**Comment by rscmbbng** [comment]
I've been thinking that it is weird that you either have `LIMITED_FEDERATION_MODE`  or a super permissive federation model, which is the default. The first basically requires one to learn of any interesting 3rd party instances outside your own instance because you can't see them. Only after discovering them through other means, can you add them to your allow list. The permissive federation model, on the other hand, basically promiscuously connects to anything. This has the opposite effect: I (unknowingly) connect to something  which I later come to regret. I then have to manually disconnect (suspend). Often, at that point, the damage is already done, as I tend to learn about these instances through a report. 

It would be great to have a ["cautiously optimistic"](https://decentpatterns.xyz/library/cautious-optimism/) federation model. This one would notify admins/mods that new servers have been discovered and that you can allow them or not. This could tie in to the neighborhood model proposed in https://github.com/hometown-fork/hometown/issues/1220, but is a different feature onto itself.

It would essentially allow you to slowly grow a list of ok servers to add to your allow list, as your instance discovers them through interactions, etc. Of course, a bonus would be to see whether the instances you are already connected to are also connecting with those new discoveries.

I don't understand why Mastodon applies this topic to Trending Topics but not to federation itself..

**Comment by dariusk** [comment]
@rscmbbng 
That's a great idea and also a really big thing to implement. This is definitely something I'd like to offer, definitely after implementing the thing this issue is about. Can you make a new issue for it?

**Comment by rscmbbng** [comment]
> That's a great idea and also a really big thing to implement. This is definitely something I'd like to offer, definitely after implementing the thing this issue is about. Can you make a new issue for it?

done!
